core/templates/domain/topic/topic-update.service.ts

```diff
@@ -446,7 +446,7 @@ angular.module('oppia').factory('TopicUpdateService', [
             var thumbnailFilename = (
               _getNewPropertyValueFromChangeDict(changeDict));
             subtopic.setThumbnailFilename(thumbnailFilename);
-          }, function(changeDict, subtopic) {
+          }, function(changeDict, topic) {
```

Comment:
Why this change? Ditto below.

Also shouldn't it cause tests to fail?

Reply:
Tests were failing for these two functions but not for the `setSubtopicTitle` function. I saw that `setSubtopicTitle` calls `subtopic.setTitle(oldTitle);` in the undo function rather than `topic.setTitle(oldTitle)`. I then made the changes above to 
`setSubtopicThumbnailFilename` and `setSubtopicThumbnailBgColor`, which made them pass the undo/redo test.

After digging deeper, my understanding is that the reason that this happens is that the undo callback returns the topic as the second parameter, rather than the subtopic we want to reverse changes to. Here is a trace of how the change object is made:

Following `setSubtopicThumbnailFilename/setSubtopicThumbnailBgColor`, the change is constructed with `_applySubtopicPropertyChange`, which calls `_applyChange` with the parameter `entity` equal to `topic`, not `subtopic`. `UndoRedoService.applyChange` is then called with the topic as `domainObject`.

The function signature of the undo function is:
```
// 'applyChangeToObject' and 'reverseChangeToObject' are functions whose
// return type depends upon the arguments passed to the constructor.
reverseChange(domainObject: any): void {
  this._reverseChangeToObject(this._backendChangeObject, domainObject);
}
```

This is the key part that stood out to me: 
`return type depends upon the arguments passed to the constructor`
So the undo callback is called with the domainObject, which we constructed as the topic, not the subtopic.

After making the changes above, it passes the undo/redo test.

I'm not sure if this requires another issue, and apologies for not including explanation with the PR.

Reply:
Ah yes, I took a closer look and I agree with this change. Thanks @shavavo!

Reply:
Done